src/pages/EnablePayments/index.js

```diff
@@ -23,12 +22,19 @@ const defaultProps = {
 };
 
 class EnablePaymentsPage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            loading: true,
+        };
+    }
+
     componentDidMount() {
-        fetchUserWallet();
+        fetchUserWallet().then(() => this.setState({loading: false}));
     }
 
     render() {
-        if (_.isEmpty(this.props.userWallet)) {
```

Comment:
Do we not check for this anymore?

Reply:
Yes, we using the state variable `loading` now.

Reply:
Agreed, but will we show this component when the prop `userWallet` is not empty?

Reply:
But there is an issue with this component which I have shared on slack. 
this component will not update if `userWallet` data never changes which will be the case most of the time.
This issue is coming from Onyx changes. I am going to raise this issue now.

Reply:
e.g. I think if a user has a wallet and they go to /enable-payments - I think we should not go into this if block, but maybe I am mistaken?

Reply:
So the Approach here is that We don't use the stored userWallet data and always request new.

```
  userWallet: {
        key: ONYXKEYS.USER_WALLET,

        // We want to refresh the wallet each time the user attempts to activate the wallet so we won't use the
        // We want to refresh the wallet each time the user attempts to activate the wallet so we won't use the
        // stored values here.

        initWithStoredValues: false,
    },
```